Code/src/main/java/nl/utwente/group10/ui/components/Block.java

```diff
@@ -0,0 +1,22 @@
+package nl.utwente.group10.ui.components;
+
+import javafx.scene.layout.StackPane;
+
+/**
+ * Base UI Component that other visual elements will extend from.
+ * If common functionality is found it should be refactored to here.
+ */
+public class Block extends StackPane {
+
+	/** Selected state of this Block*/
+	private boolean isSelected = false;
+	
+	/**
+	 * Set the selected boolean state of this Block
+	 * @param selectedState
+	 */
+		public void setSelected(boolean selectedState){
+			//TODO If another object is selected then deselect it first!!
```

Comment:
Very true, but I was specifically referring to the TODO: would the setSelected method deselect another object, and how would it find that other object? Or should this be in the documentation, i.e. should someone else who calls this method call setSelected on the other object first?